src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -176,18 +178,28 @@ private static GitHubSCMSource getSCMSource(final SCMSourceOwner scmSourceOwner)
      * With this listener one notifies to GitHub when a Job (associated to a PR) has been scheduled.
      * Sends: GHCommitState.PENDING
      */
-    @Extension public static class PRJobScheduledListener extends QueueListener {
+    @Extension
+    public static class PRJobScheduledListener extends QueueListener {
 
         /**
          * Manages the GitHub Commit Pending Status.
          */
-        @Override public void onEnterWaiting(Queue.WaitingItem wi) {
-            if (wi.task instanceof Job) {
-                Job<?,?> job = (Job) wi.task;
-                // TODO would actually be better to use GitHubSCMSource.retrieve(SCMHead, TaskListener) (when properly implemented).
-                // That would allow us to find the current head commit even for non-PR jobs.
-                SCMHead head = SCMHead.HeadByItem.findHead(job);
-                if (head instanceof PullRequestSCMHead) {
+        @Override
+        public void onEnterWaiting(Queue.WaitingItem wi) {
+            if (!(wi.task instanceof Job)) {
+                return;
+            }
+            final long taskId = wi.getId();
+            final Job<?,?> job = (Job) wi.task;
+            // TODO would actually be better to use GitHubSCMSource.retrieve(SCMHead, TaskListener) (when properly implemented).
+            // That would allow us to find the current head commit even for non-PR jobs.
+            final SCMHead head = SCMHead.HeadByItem.findHead(job);
+            if (!(head instanceof PullRequestSCMHead)) {
+                return;
+            }
+            Computer.threadPoolForRemoting.submit(new Runnable() {
```

Comment:
Why it called `Computer`? I thought it designed only for `remoting` communication for real `remoting` to slaves.